### PR TITLE
Raincatcher 33 - Implementation of checkboxfield using angular

### PR DESCRIPTION
**Pull request by spriadka**

### CODE REVIEW

**Comment by spriadka** [comment]
@secondsun . Can you look at this? (https://issues.jboss.org/browse/RAINCATCH-33)

**Comment by secondsun** [comment]
Looking, is there a cluster with a project set up I can test this on?  

If not, what versions of the various libraries do I need to have set to have the portal app work with it and I can set it up locally

File: lib/angular/fieldCheckbox/fieldCheckboxDirective.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Created by spriadka on 2/22/17.
+ */
```
Comment (secondsun) [inline lib/angular/fieldCheckbox/fieldCheckboxDirective.js]: remove the header

File: lib/angular/fieldCheckbox/fieldCheckboxController.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+/**
+ * Created by spriadka on 2/22/17.
+ */
+
```
Comment (secondsun) [inline lib/angular/fieldCheckbox/fieldCheckboxController.js]: remove the header

File: lib/angular/fieldCheckbox/fieldCheckboxValidation.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+/**
+ * Created by spriadka on 3/8/17.
+ */
+
```
Comment (secondsun) [inline lib/angular/fieldCheckbox/fieldCheckboxValidation.js]: remove the header

**Comment by secondsun** [comment]
@spriadka The code looks ok, if there is an envrionment to test in let me know, otherwise just tell me what versions of the other raincatcher modules to pull in in the portal and I will give it a spin.

File: lib/angular/template/appform-field-checkbox.tpl.html [hunk 4]
```diff
@@ -0,0 +1,19 @@
+<form name="checkboxForm" novalidate>
+    <input type="hidden" ng-model="fieldCheckboxCtrl.field.value" name="checkbox"
+           min="{{fieldCheckboxCtrl.field.fieldOptions.validation.min}}"
+           max="{{fieldCheckboxCtrl.field.fieldOptions.validation.max}}"
+           ng-model-options="{allowInvalid: true}" checkbox-validation>
+           <!-- When allowInvalid set to false, ng-model would become undefined -->
```
Comment (wtrocki) [inline lib/angular/template/appform-field-checkbox.tpl.html]: Why this is a problem? That's expected behavior.

File: lib/angular/template/appform-field-checkbox.tpl.html [hunk 5]
```diff
@@ -0,0 +1,19 @@
+<form name="checkboxForm" novalidate>
+    <input type="hidden" ng-model="fieldCheckboxCtrl.field.value" name="checkbox"
+           min="{{fieldCheckboxCtrl.field.fieldOptions.validation.min}}"
```
Comment (wtrocki) [inline lib/angular/template/appform-field-checkbox.tpl.html]: Why checkbox needs min/max validation?

**Comment by ghost** [comment]
@nialldonnellyfh @pmdarrow @paolobueno @witmicko quick review?

**Comment by emilioicai** [comment]
ping @wtrocki

**Review by emilioicai — APPROVED** [review]
Tested on cluster @vchepeli @spriadka , is this meant to be merged now?

**Pull request merged**